### PR TITLE
upgrade ember cli to 3.8 LTS

### PR DESCRIPTION
**Pull request by EndangeredMassa**
I used `ember-cli-update` (and manual changes) to upgrade this repo to ember-cli 3.4, which is the most recent LTS release.

The test suite still passes locally and I manually linked this to a local app, which also still worked.

We still need to:

- make it clear which versions of this addon work with which versions of Ember
- update the documentation examples to use modern async testing syntax https://github.com/ember-a11y/ember-a11y-testing/issues/95 / https://github.com/ember-a11y/ember-a11y-testing/pull/96

### CODE REVIEW

File: tests/integration/helpers/a11y-audit-if-test.js [hunk 1]
```diff
@@ -1,35 +1,37 @@
...
-  beforeEach() {
-    this.register('component:axe-component', Component.extend());
-  }
-});
+  test('a11yAuditIf should not execute a11yAudit', async function(assert) {
+    await render(hbs`{{#axe-component}}<button></button>{{/axe-component}}`);
 
-test('a11yAuditIf should not execute a11yAudit', function(assert) {
-  this.render(hbs`{{#axe-component}}<button></button>{{/axe-component}}`);
+    await a11yAuditIf(this.$());
```
Comment (rwjblue) [inline tests/integration/helpers/a11y-audit-if-test.js]:
```suggestion
    await a11yAuditIf(this.element);
```

File: tests/integration/helpers/a11y-audit-if-test.js [hunk 2]
```diff
@@ -1,35 +1,37 @@
...
+
+    // TODO
+    utils.getLocation = function() {
+      return {
+        search: '?enableA11yAudit=true'
+      }
+    };
+
+    try {
+      await a11yAuditIf(this.$())
```
Comment (rwjblue) [inline tests/integration/helpers/a11y-audit-if-test.js]:
```suggestion
      await a11yAuditIf(this.element)
```

File: tests/integration/helpers/a11y-audit-test.js [hunk 3]
```diff
@@ -1,66 +1,63 @@
...
-  }
-});
-
-test('a11yAudit runs successfully with jquery context', function(assert) {
-  this.render(hbs`{{#axe-component}}{{/axe-component}}`);
+  test('a11yAudit runs successfully with jquery context', async function(assert) {
+    await render(hbs`{{#axe-component}}{{/axe-component}}`);
 
-  return a11yAudit(this.$()).then(() => {
+    await a11yAudit(this.$());
```
Comment (rwjblue) [inline tests/integration/helpers/a11y-audit-test.js]: Will need to tweak this test a bit, when running without jQuery (some of the default test scenarios now run without jQuery) this whole test should be removed/guarded (so that it doesn't run when jQuery is missing).

File: tests/integration/helpers/a11y-audit-test.js [hunk 4]
```diff
@@ -1,66 +1,63 @@
...
   });
-});
 
-test('a11yAudit runs successfully with element context', function(assert) {
-  this.render(hbs`{{#axe-component}}{{/axe-component}}`);
+  test('a11yAudit runs successfully with element context', async function(assert) {
+    await render(hbs`{{#axe-component}}{{/axe-component}}`);
 
-  return a11yAudit(this.$()[0]).then(() => {
+    await a11yAudit(this.$()[0]);
```
Comment (rwjblue) [inline tests/integration/helpers/a11y-audit-test.js]:
```suggestion
    await a11yAudit(this.element);
```

File: tests/integration/helpers/a11y-audit-test.js [hunk 5]
```diff
@@ -1,66 +1,63 @@
...
 
-test('a11yAudit catches violations successfully', function(assert) {
-  this.render(hbs`{{#axe-component}}<button></button>{{/axe-component}}`);
+  test('a11yAudit catches violations successfully', async function(assert) {
+    await render(hbs`{{#axe-component}}<button></button>{{/axe-component}}`);
 
-  return a11yAudit(this.$()).catch((e) => {
-    assert.ok(e.message.startsWith('Assertion Failed: The page should have no accessibility violations. Violations:', 'error message is correct'));
+    try {
+      await a11yAudit(this.$());
```
Comment (rwjblue) [inline tests/integration/helpers/a11y-audit-test.js]:
```suggestion
      await a11yAudit(this.element);
```

File: tests/integration/helpers/a11y-audit-test.js [hunk 6]
```diff
@@ -1,66 +1,63 @@
...
-test('a11yAudit can use custom axe options', function(assert) {
-  this.render(hbs`{{#axe-component}}<button></button>{{/axe-component}}`);
+  test('a11yAudit can use custom axe options', async function(assert) {
+    await render(hbs`{{#axe-component}}<button></button>{{/axe-component}}`);
 
-  return a11yAudit(this.$(), {
-    rules: {
-      'button-name': {
-        enabled: false
+    await a11yAudit(this.$(), {
```
Comment (rwjblue) [inline tests/integration/helpers/a11y-audit-test.js]:
```suggestion
    await a11yAudit(this.element, {
```

File: tests/integration/helpers/a11y-audit-test.js (hunk 3 above)
Comment (MelSumner) [reply tests/integration/helpers/a11y-audit-test.js]: @rwjblue I added a guard- can you check and see if it's correct?

**Comment by MelSumner** [comment]
@EndangeredMassa I updated to 3.8 which is now latest LTS

**Comment by MelSumner** [comment]
after merging this, we'll need to do a major version bump.

**Comment by MelSumner** [comment]
@rwjblue do you think this is okay to merge now?

File: tests/integration/instance-initializers/axe-component-test.js [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
 import { initialize } from 'dummy/instance-initializers/axe-component';
 import {
   initialize as initializeViolationsHelper
 } from 'dummy/instance-initializers/violations-helper';
-import { moduleForComponent, test } from 'ember-qunit';
-import hbs from 'htmlbars-inline-precompile';
-import wait from 'ember-test-helpers/wait';
 
 const {
   Logger
```
Comment (rwjblue) [inline tests/integration/instance-initializers/axe-component-test.js]: We need to remove the use of `Ember.Logger`:

* https://emberjs.github.io/rfcs/0297-deprecate-ember-logger.html
* https://deprecations.emberjs.com/v3.x#toc_ember-console-deprecate-logger

Using `Ember.Logger` will emit warnings/deprecations since Ember 3.2.

---

This can (and probably should) be handled in a separate PR, I just noticed it while reviewing here...

File: tests/unit/instance-initializers/violations-helper-test.js [hunk 8]
```diff
@@ -1,22 +1,35 @@
...
 
-  afterEach() {
     sandbox.restore();
-  }
-});
+  });
 
-test('initializer sets a violationsHelper in the global scope', function(assert) {
-  initialize(application);
+  // Replace this with your real tests.
```
Comment (rwjblue) [inline tests/unit/instance-initializers/violations-helper-test.js]: This comment probably should be removed (the test below does have a valid assertion)

File: tests/unit/utils/concurrent-axe-test.js [hunk 9]
```diff
@@ -1,72 +1,72 @@
 /* global sinon, axe */
-import { module, test } from 'ember-qunit';
-import wait from 'ember-test-helpers/wait';
+import { module, test } from 'qunit';
 import { ConcurrentAxe } from 'ember-a11y-testing/utils/concurrent-axe';
+import wait from 'ember-test-helpers/wait';
```
Comment (rwjblue) [inline tests/unit/utils/concurrent-axe-test.js]: We should be using `import { settled } from '@ember/test-helpers` instead of `wait`

File: tests/integration/helpers/a11y-audit-test.js (hunk 3 above)
Comment (MelSumner) [reply tests/integration/helpers/a11y-audit-test.js]: oops. I accidentally put it in the wrong place. Fixed it!

File: tests/integration/instance-initializers/axe-component-test.js (hunk 7 above)
Comment (MelSumner) [reply tests/integration/instance-initializers/axe-component-test.js]: https://github.com/ember-a11y/ember-a11y-testing/issues/127

**Comment by drewlee** [comment]
These changes are preventing https://github.com/ember-a11y/ember-a11y-testing/pull/123 from moving further. I don't have full context, but @MelSumner, @EndangeredMassa, is this stable to merge with the main branch?

**Comment by SergeAstapov** [comment]
@MelSumner @EndangeredMassa @drewlee created #138 with upgrade to 3.12. Happy to help push it over the line
